### PR TITLE
refactor: rework tag service

### PR DESCRIPTION
**Pull request by rvolosatovs**
- Addressed comments from #38 
- Fix name parsing from the path
- Remove redundant trait implementations, constants and types
- Rely on `Content-Type` header for tag parsing
- Use `String` internally as tag name
- Validate tag names
- Improve test coverage
- Remove storage test (will be reworked in upcoming iterations)
- Define JOSE MIME type as exported constant
- Allow custom fields within `Entry`

### CODE REVIEW

File: crates/tags/src/lib.rs [hunk 1]
```diff
@@ -25,44 +28,321 @@ impl<T: Clone + Storage> From<T> for Service<T> {
...
+
+    #[async_std::test]
+    async fn name_from_request() {
+        fn new_request(path: impl AsRef<str>) -> Request {
+            let mut req = Request::new(Method::Put, "https://example.com/");
+            req.url_mut().set_path(path.as_ref());
+            req
+        }
+
+        for path in ["/", "//", "/\\/", "//test", "/=/", "/ы", "/?"] {
```
Comment (rjzak) [inline crates/tags/src/lib.rs:153]: What's the `ы` for?

Comment (rvolosatovs) [reply crates/tags/src/lib.rs:153]: Testing with various invalid input values and making sure nothing weird happens

**Comment by rjzak** [comment]
Looks good to me. I'm sure commit-flattening will be requested.

**Comment by rvolosatovs** [comment]
> Looks good to me. I'm sure commit-flattening will be requested.

Not sure what you mean.
Please approve, if you agree with the changes

**Comment by rjzak** [comment]
>> Looks good to me. I'm sure commit-flattening will be requested.

> Not sure what you mean.

I mean I think Nathaniel wants commit flattened. I'm not sure of that applies here. And I suppose I have to look around for an "Approve" button somewhere...

**Review by rjzak — APPROVED** [review]
Looks good to me.

**Pull request merged**